src/Controls/src/Core/ContentPage.cs

```diff
@@ -8,7 +8,10 @@ public partial class ContentPage : TemplatedPage
 		public View Content
 		{
 			get { return (View)GetValue(ContentProperty); }
-			set { SetValue(ContentProperty, value); }
+			set {
+				SetValue(ContentProperty, value);
+				TypeHashCode = null;
+			}
```

Comment:
This should probably be set in the BindingProperty's onChanged delegate - otherwise this will not fire when set via XAML.